src/sources/soundsourceffmpeg.cpp

```diff
@@ -67,7 +67,9 @@ SINT getSamplingRate(AVStream* pStream) {
 inline
 SINT getFrameCount(AVStream* pStream) {
     // NOTE(uklotzde): Use 64-bit integer calculation to minimize rounding errors
-    return (static_cast<uint64_t>(pStream->time_base.num) * static_cast<uint64_t>(pStream->duration) * static_cast<uint64_t>(getSamplingRate(pStream))) /
+    return (static_cast<uint64_t>(pStream->time_base.num) *
+            pStream->duration *
+            static_cast<uint64_t>(getSamplingRate(pStream))) /
             static_cast<uint64_t>(pStream->time_base.den);
```

Comment:
There is still an issue. duration is int64_t and will be converted silently to an unsigned, loosing the sign. 
We should make sure the compile uses signed calculations. It is also better to multiply a 64 bit value with a 32 bit value following the implicite conversion rules. 

How about this: 
```
return pStream->duration * getSamplingRate(pStream) * pStream->time_base.num / 
        pStream->time_base.den
```
or more obvious and save:
```
int64_t ret = pStream->duration * getSamplingRate(pStream); 
ret *=  pStream->time_base.num;
ret /= pStream->time_base.den;
if (ret > INT_MAX || ret < 0) return -1; 
return ret;
```
This will reject files longer than 13.5 h if we check for -1 later. 

Or we could move to double for FrameCount. Than we get rounding issues on long streams instead of rejecting them. 

We have a similar issue here: 
https://github.com/mixxxdj/mixxx/blob/7c58008d3ff35026763fdf7198905b068435e07e/src/sources/soundsourcesndfile.cpp#L57
where a 64 bit value is silently truncated.